src/io_proto.c

```diff
@@ -499,12 +499,13 @@ static int32_t io_encode_open_request(
 )
 {
     int32_t result;
-    __io_encode_type_begin(ctx, request, 2);
+    __io_encode_type_begin(ctx, request, 3);
```

Comment:
It is difficult to see, where the magic 3 (or 2 in the old version) comes from.
I would like to have a comment, which says what is counted into 3.